internal/client/main.go

```diff
@@ -35,10 +37,89 @@ func run(cfg *config) error {
 	}
 	defer conn.Close()
 	client := echo.NewEchoServiceClient(conn)
-	resp, err := client.Hello(context.Background(), &echo.HelloRequest{Message: cfg.Message})
+	switch cfg.Stream {
+	case "none":
+		return runUnary(client, cfg)
+	case "client":
+		return runClientStream(client, cfg)
+	case "server":
+		return runServerStream(client, cfg)
+	case "bidi":
+		return runBiDiStream(client, cfg)
+	}
+	return nil
+}
+
+func runUnary(client echo.EchoServiceClient, cfg *config) error {
+	if len(cfg.Messages) > 1 {
```

Comment:
up for debate, but imo all the flag validation logic belongs in AfterApply.

Reply:
as i reviewed this, i knew you were going to say this. so i assume you know my response?

Reply:
if the program were larger and/or there were more flags/options, then i would consider using afterapply. however, splitting that logic out i feel makes it read worse. i like linear when you can do it.